provider/pkg/provider/helm_release.go

```diff
@@ -674,6 +682,15 @@ func (r *helmReleaseProvider) Diff(ctx context.Context, req *pulumirpc.DiffReque
 
 	// Extract old inputs from the `__inputs` field of the old state.
 	oldInputs, _ := parseCheckpointRelease(olds)
+
+	// if the checkpointed inputs have no checksum, then assume no change since checksums were introduced,
+	// to avoid making a spurious release.
+	checksum, ok := oldInputs["checksum"]
+	if !ok || (checksum.IsString() && checksum.StringValue() == "") {
+		oldInputs["checksum"] = news["checksum"]
+	}
```

Comment:
This block of code is intended to avoid causing a spurious upgrade of the Helm release, as a side-effect of updating the pulumi-kubernetes dependency in an existing stack. Without this, one sees a change to the `checksum` property (was previously unset). I haven't found a good way to write a test for this, and I tested it manually.